apps/nextjs-website/src/app/[productSlug]/guides/[...productGuidePage]/page.tsx

```diff
@@ -42,6 +47,30 @@ type ProductGuidePageProps = {
   bodyConfig: ParseContentConfig;
 } & LayoutProps;
 
+export async function generateMetadata({
+  params,
+}: {
+  params: Params;
+}): Promise<Metadata> {
+  const guideProps = await getGuide(
+    params?.productSlug,
+    params?.productGuidePage ?? ['']
+  );
+
+  const { page } = guideProps;
+
+  const title = getTitleFromMarkdown(page.body);
+
+  return {
+    title,
+    openGraph: {
+      title,
+      url: page.path,
+    },
+    twitter: getTwitterMetadata(title),
+  };
```

Comment:
Is really necessary to extract the title from the Markdown format?

Wouldn't be easier to do this:
```suggestion
  const {
    page: { path, title },
  } = guideProps;

  return {
    title,
    openGraph: {
      title,
      url: path,
    },
    twitter: getTwitterMetadata(title),
  };
```
the `getGuide` should return the parsed information where you can get the page's title

Reply:
Yeah, you're right. I didn't see the title field on the `getGuide` function, but there is, thanks